examples/mongodb/README.md

```diff
@@ -1,7 +1,71 @@
 This example is desgined to be run on a mongo replicaset.
 
- * Deploy 5 mongodb units
- * Run main.go specifying the urls to use
- * It will tell you which url is the primary node
- * Use part.sh to cause a partition around the specified unit while main.go is running. Set Primary to the primary unit (ip address output from main.go)
- * See if the expected len is different from what is found in the db.
+### Deploy 5 mongodb units
+
+juju deploy mongodb -n 5
+
+### Craft a connection URL
+
+* Grab the IP address of one of the MongoDB units.
+* Get the replica set name of the cluster. Default is "myset".
+* Use <ip>:<port>?replicaSet=<replica set name> for the URL in the following steps.
+
+Example:
+
+* mongodb/2 is 10.0.3.166
+* replica set name is "jupsen"
+* URL is 10.0.3.166:27017?replicaSet=jupsen
+
+The `replicaSet` name is necessary for mgo to discover the other replica set
+members and fail over the primary during the following tests. Without it, the
+client will hang for the duration of the partition -- it won't switch to a
+newly elected primary.
+
+### Run the mongodb updates
+
+```
+go run main.go -url <URL>
+```
+
+See `go run main.go -help` for options.
+
+This will display the primary IP address. Cross-reference with `juju status` to
+find the primary unit.
+
+#### Find the current primary
+
+You can also check the current primary while the test is running, if you've
+created a partition, for example, with:
+
+```
+go run primary.go -url <URL>
+```
+
+Match the IP with `juju status` to find the primary unit.
```

Comment:
After this branch lands can you add an issue to allow primary.go to output the ip address and unit id of the primary